lib/avro-patches/logical_types/schema.rb

```diff
@@ -57,8 +57,8 @@ def self.real_parse(json_obj, names=nil, default_namespace=nil)
   end
 
   # Determine if a ruby datum is an instance of a schema
-  def self.validate(expected_schema, logical_datum, encoded = false)
-    Avro::SchemaValidator.validate!(expected_schema, logical_datum, encoded)
+  def self.validate(expected_schema, logical_datum, recursive = true, encoded = false)
```

Comment:
Should we add `recursive` as the last arg so this doesn't break the current interface? Maybe this is where things get a bit muddled with the multiple levels of patching?

Should we use an `options` hash instead to make call sites a little more obvious?